unity/Assets/Scripts/ArmAgentController.cs

```diff
@@ -127,37 +149,176 @@ public class ArmAgentController : PhysicsRemoteFPSAgentController {
+            MoveAhead(moveMagnitude: moveMagnitude, fixedDeltaTime: Time.fixedDeltaTime);
+        }
+
+        public void MoveAhead(
+            float? moveMagnitude = null,
+            float speed = 1,
+            float? fixedDeltaTime = null,
+            bool returnToStart = true,
+            bool disableRendering = true
+        ) {
+            MoveAgent(
+                ahead: moveMagnitude.GetValueOrDefault(gridSize),
+                speed: speed,
+                fixedDeltaTime: fixedDeltaTime,
+                returnToStart: returnToStart,
+                disableRendering: disableRendering
+            );
+        }
+
+        public override void MoveBack(
+            float? moveMagnitude = null,
+            string objectId = "",
+            float maxAgentsDistance = -1f,
+            bool forceAction = false,
+            bool manualInteract = false,
+            bool allowAgentsToIntersect = false
+        ) {
+            // TODO: This is an ugly necessity until we refactor all
+            // of the agent controllers
+            MoveBack(moveMagnitude: moveMagnitude, fixedDeltaTime: Time.fixedDeltaTime);
+        }
+
+        public void MoveBack(
+            float? moveMagnitude = null,
+            float speed = 1,
+            float? fixedDeltaTime = null,
+            bool returnToStart = true,
+            bool disableRendering = true
+        ) {
+            MoveAgent(
+                ahead: -moveMagnitude.GetValueOrDefault(gridSize),
+                speed: speed,
+                fixedDeltaTime: fixedDeltaTime,
+                returnToStart: returnToStart,
+                disableRendering: disableRendering
+            );
+        }
+
+        public override void MoveRight(
+            float? moveMagnitude = null,
+            string objectId = "",
+            float maxAgentsDistance = -1f,
+            bool forceAction = false,
+            bool manualInteract = false,
+            bool allowAgentsToIntersect = false
+        ) {
+            // TODO: This is an ugly necessity until we refactor all
+            // of the agent controllers
+            MoveRight(moveMagnitude: moveMagnitude, fixedDeltaTime: Time.fixedDeltaTime);
         }
 
-        public override void MoveRight(ServerAction action) {
-            throw new InvalidOperationException("When using the arm, please call controller.step(action=\"MoveAgent\", ahead=0, right=X).");
+        public void MoveRight(
+            float? moveMagnitude = null,
+            float speed = 1,
+            float? fixedDeltaTime = null,
+            bool returnToStart = true,
+            bool disableRendering = true
+        ) {
+            MoveAgent(
+                right: moveMagnitude.GetValueOrDefault(gridSize),
+                speed: speed,
+                fixedDeltaTime: fixedDeltaTime,
+                returnToStart: returnToStart,
+                disableRendering: disableRendering
+            );
         }
```

Comment:
(cc @erick) Won't this break the action dispatcher? Read:
https://github.com/allenai/ai2thor/blob/f49351084337547964732c00f8cc96af98182380/unity/Assets/Scripts/ActionDispatcher.cs#L46

My original idea was to just have
```c#
        public override void MoveRight(
            float? moveMagnitude = null,
            string objectId = "",
            float maxAgentsDistance = -1f,
            bool forceAction = false,
            bool manualInteract = false,
            bool allowAgentsToIntersect = false
        ) {
            ...
        }
```
and basically ignore `allowAgentsToIntersect`.

But it seems dangerous to have an ambiguous action with:
```c#
        public override void MoveRight(
            float? moveMagnitude = null,
            string objectId = "",
            float maxAgentsDistance = -1f,
            bool forceAction = false,
            bool manualInteract = false
        ) {
            ...
        }

        public override void MoveRight(
            float? moveMagnitude = null,
            string objectId = "",
            float maxAgentsDistance = -1f,
            bool forceAction = false,
            bool manualInteract = false,
            bool allowAgentsToIntersect = false
        ) {
            ...
        }
```

Reply:
Ok I've made a change suggested by @ekolve: creating single functions that contain all parameters from the default and arm agent controllers. This is a bit ugly but it is unambiguous and we can can remove parameters when we eventually do a refactor.